src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -211,6 +211,67 @@ def parse_destination(dest_str):
     return try_call(resolve_path, dest_str)
 
 
+def _check_suggestions(src_dir, publish=False):
+    """
+    Examines the specified dxapp.json file and warns about any
+    violations of suggestions guidelines.
+    """
+    dxapp_json_filename = os.path.join(src_dir, "dxapp.json")
+    app_spec = json.load(open(dxapp_json_filename))
+    for input_field in app_spec.get('inputSpec', []):
+        for suggestion in input_field.get('suggestions', []):
+            if 'project' in suggestion:
+                try:
+                    project = dxpy.api.project_describe(suggestion['project'], {"permissions": True})
+                    if 'PUBLIC' not in project['permissions'] and publish:
+                        logger.warn('Warning, project {name} NOT PUBLIC!'.format(name=project['name']))
+                except dxpy.exceptions.DXAPIError as e:
+                    if e.code == 404:
+                        raise dxpy.app_builder.AppBuilderException(
+                            'Suggested project {name} does not exist, or not accessible by user'.format(
+                                name=suggestion['project']))
+                if 'path' in suggestion:
+                    try:
+                        check_folder_exists(suggestion['project'], suggestion['path'], '')
+                    except ResolutionError as e:
+                        raise dxpy.app_builder.AppBuilderException(
+                            'Folder {path} could not be found in project {project}'.format(
+                                path=suggestion['path'], project=suggestion['project']))
+            if '$dnanexus_link' in suggestion:
+                if suggestion['$dnanexus_link'].startswith(('file-', 'record-', 'gtable-')):
+                    try:
+                        object = dxpy.describe(suggestion['$dnanexus_link'])
```

Comment:
One last thing, please consider using a different name for the results from `describe` given that `object` is a python built-in function, and reassigning  it might have unwanted  side effects.
Also in line 266. 
Thanks